src/ldp/nn/handlers/transformer_handler.py

```diff
@@ -192,6 +194,14 @@ async def __call__(  # type: ignore[override]
     @staticmethod
     def model_generate(model: PreTrainedModel, *args, **kwargs):
         """A method that can be used as module_call_fn to sample from an LLM."""
+        if int(os.environ.get("WORLD_SIZE", "1")) > 1:
+            synced_gpus = kwargs.pop("synced_gpus", None)
+            if synced_gpus is None:
+                logger.debug("synced_gpus not defined, defaulting to True.")
+                kwargs["synced_gpus"] = True
+            elif not synced_gpus:
+                raise ValueError("synced_gpus must be True when using FSDP.")
```

Comment:
I think this is set here: https://github.com/huggingface/transformers/blob/79254c9b61ac2e1a6e0f698c38b1654159095e2e/src/transformers/generation/utils.py#L2014-L2015